renderer/components/Request/RequestSummary.js

```diff
@@ -15,180 +16,164 @@ import { CryptoSelector, CryptoValue, FiatSelector, FiatValue } from 'containers
 import { Truncate } from 'components/Util'
 import messages from './messages'
 
-class RequestSummary extends React.Component {
-  state = {
-    isExpired: null,
-    timer: null,
-  }
-
-  static propTypes = {
-    intl: intlShape.isRequired,
-    /** Lnd invoice object for the payment request */
-    invoice: PropTypes.object,
-    /** Lightning Payment request. */
-    payReq: PropTypes.string.isRequired,
-    /** Show a notification. */
-    showNotification: PropTypes.func.isRequired,
-  }
+const RequestSummary = ({ invoice = {}, payReq, intl, showNotification, ...rest }) => {
+  const [isExpired, setIsExpired] = useState()
 
-  componentDidMount() {
-    const { payReq } = this.props
+  const scheduler = createScheduler()
```

Comment:
should we use `ref` here? Probably we shouldn't re-create scheduler on every render

Reply:
Yeah, totally. I'm surprised it was working as it was tbh. I've updated and rechecked and it looks to still be working correctly.